pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -829,9 +920,158 @@ func (m *kubeGenericRuntimeManager) SyncPod(pod *v1.Pod, podStatus *kubecontaine
 		start("container", containerStartSpec(&pod.Spec.Containers[idx]))
 	}
 
+	// Step 8: For containers in podContainerChanges.ContainersToUpdate[Cpu,Memory] list, invoke UpdateContainerResources
+	if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) &&
+		(len(podContainerChanges.ContainersToUpdate) > 0 || len(podContainerChanges.ContainersToRestart) > 0) {
+		pcm := m.containerManager.NewPodContainerManager()
+		podResources := cm.ResourceConfigForPod(pod, m.cpuCFSQuota, uint64((m.cpuCFSQuotaPeriod.Duration)/time.Microsecond))
+		if podResources == nil {
+			klog.Errorf("Unable to get resource configuration for pod %s", pod.Name)
+			return
+		}
+		setPodCgroupConfig := func(rName v1.ResourceName) error {
+			var err error
+			switch rName {
+			case v1.ResourceCPU:
+				err = pcm.SetPodCgroupCpuConfig(pod, podResources.CpuQuota, podResources.CpuPeriod, podResources.CpuShares)
+			case v1.ResourceMemory:
+				err = pcm.SetPodCgroupMemoryConfig(pod, *podResources.Memory)
+			}
+			if err != nil {
+				klog.Errorf("Failed to set %s cgroup config for pod %s failed: %v", rName, pod.Name, err)
+			}
+			return err
+		}
+		// If resize results in net pod resource increase, set pod cgroup config before resizing containers.
+		// If resize results in net pod resource decrease, set pod cgroup config after resizing containers.
+		// If an error occurs at any point, abort. Let future syncpod iterations retry the unfinished stuff.
+		resizeContainers := func(rName v1.ResourceName, currPodCgLimit, newPodCgLimit int64) error {
+			var err error
+			if newPodCgLimit > currPodCgLimit {
+				if err = setPodCgroupConfig(rName); err != nil {
+					return err
+				}
+			}
+			if len(podContainerChanges.ContainersToUpdate[rName]) > 0 {
+				if err = m.updatePodContainerResources(pod, podStatus, rName, podContainerChanges.ContainersToUpdate[rName]); err != nil {
```

Comment:
Why do we need to separate the updatePodContainerResources calls for each resource?

Reply:
We may be handling a resize where we are doing a net-increase of say memory, while doing a net-decrease of CPU.

Reply:
Got it, thanks